### PR TITLE
Fixes

### LINKED ISSUE #12569: [Examples] Move and delete some modules and directories.
Hi community,

This issue is for #10887.

### Background

Refor to #12557.

### Aim
1. Move `shardingsphere-parser-example` to `other-example` .
2. Delete directory `docker/shardingsphere-jdbc`.

### How to
1. Create new module  `other-example` in `examples` project.
2. Move module `shardingsphere-parser-example` into `other-example` .
3. Delete directory `docker/shardingsphere-jdbc`.

### PR DESCRIPTION
**Pull request by Swastyy**
Fixes #12569 

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

**Comment by RaigorJiang** [comment]
Hi @Swastyy 
The modules in [pom](https://github.com/apache/shardingsphere/blob/master/examples/pom.xml) file need to update too.

**Comment by Swastyy** [comment]
> Hi @Swastyy
> The modules in [pom](https://github.com/apache/shardingsphere/blob/master/examples/pom.xml) file need to update too.

Ohh, I will change.

**Comment by Swastyy** [comment]
@RaigorJiang Do I have to update `docker` and `src` modules in pom file?

**Comment by RaigorJiang** [comment]
Sorry, there are two more questions:
1. `other-example` is a new module, it also has its own `pom` file, its parent is `shardingsphere-example`, and it has a submodule, `shardingsphere-parser-example`.
2. The parent of `shardingsphere-parser-example` should be updated to `other-example`.

File: examples/other-example/shardingsphere-parser-example/pom.xml [hunk 1]
```diff
@@ -25,7 +25,7 @@
         <artifactId>shardingsphere-example</artifactId>
```
Comment (RaigorJiang) [inline examples/other-example/shardingsphere-parser-example/pom.xml:25]: parent shoud be other-example

Comment (Swastyy) [reply examples/other-example/shardingsphere-parser-example/pom.xml]: Correct now?

**Review by RaigorJiang — APPROVED** [review]

File: examples/other-example/shardingsphere-parser-example/pom.xml (hunk 1 above)
Comment (RaigorJiang) [reply examples/other-example/shardingsphere-parser-example/pom.xml]: Yes, thank you~

**Pull request merged**